lua/cfc_prop_restoration/core/sv/sv_init.lua

```diff
@@ -287,12 +287,14 @@ function CFC_PropRestore.SaveProps()
     nextSave = time + autosaveDelay
 end
 
+hook.Add( "CFC_DailyRestart_SoftRestart", "CFC_PropRestore_SaveProps", saveProps )
+
 timer.Create( "CFC_Restoration_Think", 5, 0, function()
     local time = CurTime()
 
     -- Autosaving props
-    if time >= nextSave then
-        CFC_PropRestore.SaveProps()
+    if time >= nextSave and table.IsEmpty( queue ) then
```

Comment:
The `saveProps` function already implements the `IsEmpty` check so you can remove it here

```suggestion
    if time >= nextSave then
```